drill.cpp

```diff
@@ -847,12 +847,16 @@ map<int, drillbit> ExcellonProcessor::optimize_bits() {
                     return a.difference(wanted_length, inputFactor).value_or(std::numeric_limits<double>::infinity()) <
                         b.difference(wanted_length, inputFactor).value_or(std::numeric_limits<double>::infinity());
                 });
-            if (best_available_drill->difference(wanted_length, inputFactor)) {
+
+            auto difference = best_available_drill->difference(wanted_length, inputFactor);
+            if (difference) {
                 wanted_drill_bit.diameter = best_available_drill->diameter().asInch(inputFactor);
                 wanted_drill_bit.unit = "inch";
-                cerr << "Info: bit " << wanted_drill.first << " ("
-                   << old_string << ") is rounded to "
-                   << drill_to_string(wanted_drill_bit) << std::endl;
+                if (difference > 1e-6 || difference < -1e-6) {
```

Comment:
Or `abs(difference) > 1e-6`.  That way you don't have to repeat the constant.

Another way that we could do this PR is to string compare `old_string` with `drill_to_string(wanted_drill_bit)`.  But your way seems better, yes.

Reply:
Using `abs()` was my initial approach, but then the compiler started shouting at me:
```
drill.cpp:855:35: error: no matching function for call to ‘abs(const boost::optional<double>&)’
                 if (abs(difference) > 1e-6) {
                                   ^
[long list of "no known conversion for argument 1 from ‘const boost::optional<double>’ to '[other format]']
```

Reply:
I didn't even realize that it's an optional! Huh.

So you'll need `if (difference && abs(*difference) > 1e-6)`

Sorry about that! I must've had a good reason why it's optional but I forgot. Maybe because you might have units that can't be subtracted together or something.